controllers/drplacementcontrol_controller.go

```diff
@@ -1213,24 +1223,272 @@ func (r *DRPlacementControlReconciler) updateResourceCondition(
 		drpc.Status.LastGroupSyncTime = nil
 
-		r.setLastSyncTimeMetric(syncmetric, nil, log)
-	} else {
-		drpc.Status.ResourceConditions.ResourceMeta.Kind = vrg.Kind
-		drpc.Status.ResourceConditions.ResourceMeta.Name = vrg.Name
-		drpc.Status.ResourceConditions.ResourceMeta.Namespace = vrg.Namespace
-		drpc.Status.ResourceConditions.ResourceMeta.Generation = vrg.Generation
-		drpc.Status.ResourceConditions.Conditions = vrg.Status.Conditions
-
-		protectedPVCs := []string{}
-		for _, protectedPVC := range vrg.Status.ProtectedPVCs {
-			protectedPVCs = append(protectedPVCs, protectedPVC.Name)
+		SetDRPCStatusCondition(
+			&drpc.Status.Conditions,
+			rmn.ConditionProtected,
+			drpc.Generation,
+			metav1.ConditionUnknown,
+			rmn.ReasonProtectedUnknown,
+			fmt.Sprintf("Missing VolumeReplicationGroup status from cluster %s", clusterName))
+
+		return
+	}
+
+	drpc.Status.ResourceConditions.ResourceMeta.Kind = vrg.Kind
+	drpc.Status.ResourceConditions.ResourceMeta.Name = vrg.Name
+	drpc.Status.ResourceConditions.ResourceMeta.Namespace = vrg.Namespace
+	drpc.Status.ResourceConditions.ResourceMeta.Generation = vrg.Generation
+	drpc.Status.ResourceConditions.Conditions = vrg.Status.Conditions
+
+	protectedPVCs := []string{}
+	for _, protectedPVC := range vrg.Status.ProtectedPVCs {
+		protectedPVCs = append(protectedPVCs, protectedPVC.Name)
+	}
+
+	drpc.Status.ResourceConditions.ResourceMeta.ProtectedPVCs = protectedPVCs
+	drpc.Status.LastGroupSyncTime = vrg.Status.LastGroupSyncTime
+
+	SetDRPCProtectedCondition(drpc, vrg, clusterName)
+}
+
+// SetDRPCProtectedCondition updates the DRPC status condition Protected based on various states of VRG, from the
+// cluster where the workload is expected to be placed. The VRG passed in should be the one where the workload is
+// expected to be deployed.
+func SetDRPCProtectedCondition(
+	drpc *rmn.DRPlacementControl,
+	vrg *rmn.VolumeReplicationGroup,
+	clusterName string,
+) {
+	// If Primary only then process ClusterDataReady; as when Secondary, ClusterDataReady is a don't care!
+	if vrg.Spec.ReplicationState == rmn.Primary && ProcessVRGClusterDataReady(drpc, vrg, clusterName) {
+		return
+	}
+
+	if ProcessVRGClusterDataProtected(drpc, vrg, clusterName) {
+		return
+	}
+
+	if ProcessVRGDataReadyAndProtected(drpc, vrg, clusterName) {
+		return
+	}
+
+	// Process last sync time, if > 3x the schedule report protected as false
+
+	SetDRPCStatusCondition(&drpc.Status.Conditions, rmn.ConditionProtected, drpc.Generation,
+		metav1.ConditionTrue,
+		rmn.ReasonProtected,
+		fmt.Sprintf("VolumeReplicationGroup on cluster %s is protecting required resources and data",
+			clusterName))
+}
+
+// ProcessVRGClusterDataReady is a helper function used by SetDRPCProtectedCondition
+//   - Returns a bool that is true if status was updated, and false otherwise,
+//     requiring further processing of vrg for a status decision
+func ProcessVRGClusterDataReady(drpc *rmn.DRPlacementControl,
```

Comment:
Maybe change the function names to start with the word `Update...` instead of `Process...`
So this function becomes `UpdateVRGClusterDataProtected` which returns either `Updated` or `Not Updated`

Reply:
Done